homeassistant/components/tesla/sensor.py

```diff
@@ -87,6 +87,16 @@ def unit_of_measurement(self):
                 self._unit = LENGTH_KILOMETERS
                 self.current_value /= 0.621371
                 self.current_value = round(self.current_value, 2)
+        elif self.tesla_device.bin_type == 0xC:
```

Comment:
I think the logic to understand the hex codes (i.e. `0xC`) should be encapsulated by the library, not in HA.

Reply:
That will take a significant rewrite of the component and library.

Reply:
Device specific logic should be in the device library not in home assistant.

Reply:
@MartinHjelmare I don't disagree. However, you had asked that we submit only small PRs. Should I close this PR, rewrite the component/library to stop using bin_types and then do a new PR for this feature? 

Please remember I'm trying to maintain an older component that is effectively unmaintained. If the preference is that no new features get added until a rewrite, then I'm happy to do feature development as a custom component and only do bug fixes in the official component.

Reply:
We shouldn't extend bad patterns. If a feature doesn't do that then it's ok.

In this case the library should be extended to handle this logic before we can add that feature.

Reply:
This is just a kilometer conversion based off a GUI setting in the app. The API only returns data back in miles and the app is responsible for converting to any other unit.

Reply:
As a note, I already removed the bin_types and have another PR to remove them entirely from the component so this thread is stale.